### ISSUE TITLE
Docker Desktop for Apple Silicon fails to start with CPUS > 8 on M1 Max MBP

### ISSUE THREAD

**Issue by sharrief** [docker/for-mac#6034]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 7B1AEBCD-0482-42AD-A5CB-41BA1876B437/20211107175041

### Expected behavior
Docker Desktop starts

### Actual behavior
Docker Desktop stops and starts continuously

### Information

  - Is it reproducible? Yes
  - Is the problem new? No
  - Did the problem appear with an update? No
  - A reproducible case if this is a bug, Dockerfiles FTW.

  - macOS Version: 12.0.1
  - Intel chip or Apple chip: Apple M1 Max
  - Docker Desktop Version: 4.1.1

### Steps to reproduce the behavior
1. Restart MacOS
2. Install Docker for Desktop for Apple Silicon (install Rosetta 2, install helper, accept ToS)
3. Skip tutorial
4. Go to Settings
5. Disable weekly tips
6. Go to Resources>Advanced
7. Set CPUS to a value > 8
8. Click Apply & Restart
9. Docker Desktop stops and starts continuously
10. Reset to Factory Defaults or uninstall Docker Desktop to fix

In the video below I set CPUS to 8 just to show that up to 8 CPUS does not reproduce the issue. Guessing it has to do with the efficiency cores??

https://user-images.githubusercontent.com/17752660/140656067-30f3bdbe-1bab-4de1-9c52-27537b9b1be3.mov

**Comment by djs55** [comment]
Hi @sharrief, thanks for the issue.

Could you let me know if it behaves any differently with the experimental Virtualization framework enabled? It's in the Whale menu -> Preferences... -> Experimental Features -> Use the new Virtualization framework.

Thanks!

**Comment by 0x15f** [comment]
I was encountering the same issue (same m1 specs and steps to reproduce), enabling the new Virtualization framework did resolve the issue for me. cc @djs55

**Comment by Charles-Geometric** [comment]
Is there some way to change these settings without opening the GUI? The preferences screen simply never loads for me:

<img width="1274" alt="Screenshot 2021-11-09 at 16 42 38" src="https://user-images.githubusercontent.com/88628000/140966863-c2e8e198-dbdb-4ce5-b445-251924e01659.png">
.

**Comment by djs55** [comment]
@Charles-Geometric you can edit (from memory) `~/Library/Containers/com.docker.docker/Data/settings.json` and change `useVirtualizationFramework` to `true` and restart.

**Comment by Charles-Geometric** [comment]
> @Charles-Geometric you can edit (from memory) `~/Library/Containers/com.docker.docker/Data/settings.json` and change `useVirtualizationFramework` to `true` and restart.

There's no json file in that folder.

**Comment by Charles-Geometric** [comment]
Found it here: https://github.com/docker/for-mac/issues/5410#issuecomment-832506745

`~/Library/Group\ Containers/group.com.docker/settings.json`

It seems to solve the problem, flipping that bool.

**Comment by djs55** [comment]
@Charles-Geometric glad you found the right folder and that it worked for you! Thanks for the confirmation.

**Comment by sharrief** [comment]
> Hi @sharrief, thanks for the issue.
> 
> Could you let me know if it behaves any differently with the experimental Virtualization framework enabled? It's in the Whale menu -> Preferences... -> Experimental Features -> Use the new Virtualization framework.
> 
> Thanks!

Yep, that fixed the issue for me. I also noticed faster container startup and performance in the container. Thanks!
My issue is now resolved, though I think the constantly restarting without the new Virtualization framework is a bug.
